### ISSUE TITLE
Integrate new Hierarchical Action List with Component List Panel View

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4446]
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/181869402).
<hr />

## Why

**As** Enso IDE user
**I want** to browse Enso components in new component browser
**So that** I can look for a desired component in a convenient way

## Scope

The Component Panel List should replace the old Node Searcher. In case where we don't filter entries (the "pattern" part of input expression is empty) we should properly display the virtual groups, local variables and top-level modules. When we filter items, the modules should be flattened as described in the [design doc](https://github.com/enso-org/design/blob/main/epics/component-browser/design.md#component-module-groups-during-searching). 

What is *not* in the scope: entering submodules and integrating breadcrumbs.

## Acceptance Criteria
```gherkin
Scenario: 
GIVEN I have new project created
WHEN I open Component Browser
THEN I see:
  1. in Favourites Tools section: Virtual Component Groups from Standard Library
  2. in Local Scope section: one variable operator1
  3. in Submodules section: All level-1 modules from Standard Library
WHEN I start typing characters
THEN I see results filtered by written leters, and Submodules are flattened.
WHEN  I select an entry from A) Favourites Tools section or B) Submodules section and press TAB
THEN the entry is added to my current input, as in the old Node Searcher, and 
WHEN I clear the input, select an entry from A) Favourites Tools section or B) Submodules section and press ENTER
THEN a new node is created with selected entry input.

```

## Blockers:
[#181865548](https://www.pivotaltracker.com/story/show/#181865548) resolved
[#181869186](https://www.pivotaltracker.com/story/show/#181869186) resolved
[#182573549](https://www.pivotaltracker.com/story/show/#182573549) resolved

## Comments:
**Adam Obuchowicz** reports a new **STANDUP** for the last Friday (2022-06-03):

**Progress:** Created a feature flag for new browser and start integrating the CB stub with the ProjectView It should be finished by 2022-06-10.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Have a view embedded, and start working on presentation layer. (Enso Bot - Jun 6, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-06):

**Progress:** A little progress, due tu many meetings and stuff to think about. Updated branch to newest Michael work. It should be finished by 2022-06-10.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Have a view embedded, and start working on presentation layer. (Enso Bot - Jun 7, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-06-07):

**Summary:** There is **1 day** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 1 day delay** for the delivery of this weekly plan.

**Delay Cause:** Having both searchers in Project View turned out to be a bit complicated - I needed to think about some "API" unification, took several approaches. (Enso Bot - Jun 8, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-06-07):

**Summary:** There is **2 day** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 0 day delay** for the delivery of this weekly plan.

**Delay Cause:** Forgot about weekend (Enso Bot - Jun 8, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-07):

**Progress:** Worked on integrating the Component Panel List + Documentation view, and embedding it into Project View with old searcher. It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Finish embedding view and do some progress in presenter. (Enso Bot - Jun 8, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-08):

**Progress:** The initial integration is written but buggy. It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Debug and code cleanup. (Enso Bot - Jun 9, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-09):

**Progress:** All issues were fixed, but could not progressed with documentation/acceptance integration due to missing feature in the view. It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). If agreed with Michael, I will add support for selecting/accepting items in Component List Panel, and then integrate. (Enso Bot - Jun 10, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-06-13):

**Summary:** There is **4 day** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 0 day delay** for the delivery of this weekly plan.

**Delay Cause:** 1. Some time was spent on improving View, and there is possibility that I will help in View a bit more
2. The Holiday on Thursday (Enso Bot - Jun 14, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-13):

**Progress:** Suggestion acceptance (tab key) integrated, expression acceptance in progress. It should be finished by 2022-06-17.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Finish expression acceptance and documentation integration. Fixing issues. (Enso Bot - Jun 14, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-14):

**Progress:** The accepting expression and documentation works. It should be finished by 2022-06-17.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). There are still some performance issues to fix + code cleanup. (Enso Bot - Jun 15, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for the last Friday (2022-06-17):

**Summary:** There is **7 day** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 5 days delay** for the delivery of this weekly plan.

**Delay Cause:** There were many issues spotted immediately by me during testing the integration. Also, the delay is so big, because the integration branch is blocked by integration of selection box and finishing the View. (Enso Bot - Jun 20, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for the last Friday (2022-06-17):

**Progress:** Cleaned up the code and fix some bugs. However, the performance issues are harder than I thought. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). There is one problem with filtering to fix. Also I will think how to optimize the View basing on Mateusz's measurements. (Enso Bot - Jun 20, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for the provided date (2022-06-17):

**Progress:** Helping AO with integration: started profiling unexpected slowness of the CB when writing in the Expression Input. Most of the time was spent trying to build & dump a performance profile (had issues with the build script & with unexpectedly obsolete profiling instructions). By the end of the day, started doing actual profiling by iteratively adding profiling spans to "dig down" in the flame graph. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Continue profiling. (Enso Bot - Jun 20, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-06-20):

**Progress:** Continued profiling slowness of CB. Shared the raw results on #ide and discussed the observations on a call with AO. It should be finished by 2022-06-24. (Enso Bot - Jun 21, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-21):

**Progress:** Most of the time spent on planning new Component View using three lists, which will be implemented in a new User Story. Debugged a bit more problems with filtering. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Track down all issues (except performance). If everything go good, the PR will be created. (Enso Bot - Jun 22, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-22):

**Progress:** Many small fixes delivered to branch. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Help with merging blocking PRs and create a PR to merge the integration. (Enso Bot - Jun 23, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-06-27):

**Summary:** There is **4 days** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** The Integration waits for Michaels PR, and Michael is off today. I'm able to work on other (desing) issues, however. (Enso Bot - Jun 28, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-06-27):

**Progress:** Did the integration of Mateusz branch with local scope entries, resolve conflicts. Discovered more issues with filtering + zooming out scene. It should be finished by 2022-06-28.

**Next Day:** Next day I will be working on the [##182560195 task](https://www.pivotaltracker.com/story/show/182560195). besides the occasional fixes for integration branch (which is blocked anyway by the second View PR), I will start designing my user stories. (Enso Bot - Jun 28, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-06-28):

**Progress:** Fixes in local scope entries integration It should be finished by 2022-06-28. (Enso Bot - Jun 28, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-06-28):

**Summary:** There is **3 days** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 1 days delay** for the delivery of this weekly plan.

**Delay Cause:** Two days are previous dealy's forgotten weekend, one is for fixing sublayers which is non-trivial (Enso Bot - Jun 29, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-06-29):

**Progress:** Found a reliable reproduction scenario for Michael, to show a bug in CB view where the groups does not disappear. Debugged the problem with groups zoomed out. Discussed with Wojciech one solution, where sublayers "inherits" camera from its parents. Some details are not still clear, however. It should be finished by 2022-07-01.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). fix the bug, help Michael in his bugs and check again the integration for more bugs (Enso Bot - Jun 29, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-06-30):

**Progress:** Progressing with replacing selection layer was hard; created a new branch, where layers can have only one parent. However, it is not clear what to do with mask layers. Wojciech promised to check this and give me info what to do with it. It should be finished by 2022-07-01. (Enso Bot - Jun 30, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for the last Friday (2022-07-01):

**Summary:** There is **5 days** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 2 days delay** for the delivery of this weekly plan.

**Delay Cause:** Still waiting for Michaels PR. Also the proper fix for sublayer's cameras seems to be very non-trivial. (Enso Bot - Jul 4, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for the last Friday (2022-07-01):

**Progress:** Making sublayers keep only one parent layer. Also needed to replace it with full WeakLayer, to be able to obtain camera. The refactoring is almost done, but the debug scene does not work. It should be finished by 2022-07-06. (Enso Bot - Jul 4, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-07-04):

**Progress:** Applied a workaround for the problem with camera. Cleaned up the branch; fixed linter warnings, reformatted and added a changelog entry. It should be finished by 2022-07-06.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Do self-review and request a review. (Enso Bot - Jul 4, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-07-05):

**Progress:** There were many fixes required: one test failed and the fix was not trivial to implement. It should be finished by 2022-07-06.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). fix all last fixes and create PR (Enso Bot - Jul 6, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-07-06):

**Progress:** Fixed all problems and un-drafted PR It should be finished by 2022-07-06. (Enso Bot - Jul 6, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-07-11):

**Summary:** There is **9 days** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 2 days delay** for the delivery of this weekly plan.

some time was just waiting for QA (I was only one from the team on Friday). As there are many issues, I extend more time for fixing them - perhaps some will wait for Ilya which comes back on Thursday

**Delay Cause:** There were many issues reported during QA process, not all are trivial to fix

**Possible solutions:** The PR is just a big deilvery (Enso Bot - Jul 12, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-07-11):

**Progress:** Fixed some issues found by Accetance process. It should be finished by 2022-07-15.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Fix as much as I can from rest of the issues. (Enso Bot - Jul 12, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-11):

**Progress:** Started doing Acc/QA Review. Submitted an initial set of issues found (ones I noticed early and that were easy to reproduce deterministically). It should be finished by 2022-07-15.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181869402). Continue working on Acc/QA Review: check conformance with Design Doc; try to reproduce some nondeterministic issues I observed during the initial testing. (Enso Bot - Jul 12, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-07-12):

**Summary:** There is **2 days** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 2 days delay** for the delivery of this weekly plan.

**Delay Cause:** I've got a bit distracted by checking the CB integration PR issues found by QA. (Enso Bot - Jul 13, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-12):

**Progress:** Continued doing Acc/QA Review. Reported next set of 12 potential issues observed. [Note: The most recent delay report by AO should be cancelled - it was submitted accidentally to a wrong ticket. However, I can't do it currently through the EnsoBot - it replies with an error. I reported an EnsoBot issue on Discord.] It should be finished by 2022-07-17. (Enso Bot - Jul 13, 2022)
<hr />
**Acceptance/QA** review set to **revise**

See the list of issues in https://github.com/enso-org/enso/pull/3530#pullrequestreview-1036105149 and in the earlier comments on the PR. (Mateusz Czapliński - Jul 14, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-13):

**Progress:** Final Acc/QA Review checks. No more issues found. Marked for Revise based on the issues found on previous days. It should be finished by 2022-07-17. (Enso Bot - Jul 14, 2022)
<hr />
**Acceptance/QA** review set to **pass**

All issues noticed in the previous review were either fixed, or rejected as not in scope, or postponed and extracted to separate tasks. Thus marking the Review as Passed, as agreed with @farmaazon . For a detailed list of the status of each issue, see: https://github.com/enso-org/enso/pull/3530#issuecomment-1187676313 (Mateusz Czapliński - Jul 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-07-18):

**Summary:** There is **1 day** delay in implementation of the [Integrate new Hierarchical Action List with Component List Panel View  (#181869402) task](https://www.pivotaltracker.com/story/show/181869402).
It **will cause 1 day delay** for the delivery of this weekly plan.

**Delay Cause:** Fairly big integration task, resulting in many issues being found and surfacing for the first time. Tracking the status of all those issues introduced a noticeable process overhead. (Enso Bot - Jul 19, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-18):

**Progress:** Final summary round of Acc/QA Review. Tracked the status of all the issues found in the PR and summarized it in a comment in the PR. Created a few more tickets in Pivotal Tracker for issues that "slipped through the cracks". Discussed with AO and marked the review as Pass due to all issues being either fixed, dismissed as not in scope, or moved to separate tickets in Pivotal for future planning. It should be finished by 2022-07-18. (Enso Bot - Jul 19, 2022)
<hr />
